specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json

```diff
@@ -0,0 +1,1430 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Microsoft Cognitive Language Service - Analyze Conversations",
+    "description": "This API accepts a request and mediates among multiple language projects, such as LUIS Generally Available, Question Answering, Conversation, and then calls the best candidate service to handle the request. At last, it returns a response with the candidate service's response as a payload.\n\n In some cases, this API needs to forward requests and responses between the caller and an upstream service.",
+    "version": "2022-03-01-preview"
+  },
+  "securityDefinitions": {
+    "apim_key": {
+      "type": "apiKey",
+      "description": "A subscription key for a Language service resource.",
+      "name": "Ocp-Apim-Subscription-Key",
+      "in": "header"
+    }
+  },
+  "security": [
+    {
+      "apim_key": []
+    }
+  ],
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{Endpoint}/language",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "$ref": "common.json#/parameters/Endpoint"
+      }
+    ]
+  },
+  "paths": {
+    "/:analyze-conversations": {
+      "post": {
+        "operationId": "ConversationAnalysis_AnalyzeConversation",
+        "description": "Analyzes the input conversation utterance.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
```

Comment:
This is a breaking change from 2021-11-01-preview, in which you had:

```json
        "parameters": [
          {
            "$ref": "common.json#/parameters/ProjectNameQueryParameter"
          },
          {
            "$ref": "common.json#/parameters/DeploymentNameQueryParameter"
          },
          {
            "$ref": "#/parameters/ConversationAnalysisOptions"
          },
          {
            "$ref": "common.json#/parameters/ApiVersionParameter"
          }
        ],
```

Was this intentional?

Reply:
@annatisch this may mean that we have to conditionally format requests based on the api-version.

Reply:
I see the only addition was a `verbose` property to the now-in-body parameters. Was there another reason for moving to an in-body input?

Reply:
Yes. This is a new design and you will find that TA has the same design.

Reply:
> This is a breaking change from 2021-11-01-preview, in which you had:
> 
> ```json
>         "parameters": [
>           {
>             "$ref": "common.json#/parameters/ProjectNameQueryParameter"
>           },
>           {
>             "$ref": "common.json#/parameters/DeploymentNameQueryParameter"
>           },
>           {
>             "$ref": "#/parameters/ConversationAnalysisOptions"
>           },
>           {
>             "$ref": "common.json#/parameters/ApiVersionParameter"
>           }
>         ],
> ```
> 
> Was this intentional?

Yes. This is a new design and you will find that TA has the same design.

Reply:
> I see the only addition was a `verbose` property to the now-in-body parameters. Was there another reason for moving to an in-body input?

I think this is the new design, projectName and deployment are not in the request boby instead of query parameters.

Reply:
I think you mean *are in* the request body, but is still confirmation this was intentional. Just wanted to make sure. We'll definitely need to make some conditional changes to the SDKs.

/cc @annatisch @AhmedLeithy @ZulaMostafa @mshaban-msft